doc/source/user/tutorial-ma.rst

```diff
@@ -0,0 +1,374 @@
+functions and operations on masked data. The output is then a masked array.
+We'll see some examples of how this works in practice below.
+
+**Using masked arrays to see COVID-19 data**
+
+From `Kaggle <https://www.kaggle.com/atilamadai/covid19>`_ it is possible to
+download a dataset with initial data about the COVID-19 outbreak in the
+beginning of 2020. We are going to look at a small subset of this data,
+contained in the file ``who_covid_19_sit_rep_time_series.csv``.
+
+.. code::
+   
+    >>> import numpy as np
+    >>> filename = "who_covid_19_sit_rep_time_series.csv"
+
+Let's explore the data inside this file for the first 14 days of records. To
+gather data from the ``.csv`` file, we will use the :func:`numpy.genfromtxt`
+function, making sure we select only the columns with actual numbers instead of
+the first three columns which contain location data. We also skip the first 7
+rows of this file, since they contain other data we are not interested in.
+Separately, we will extract the information about dates and location for this
+data. 
+
+.. code::
+   
+    >>> nbcases = np.genfromtxt(filename, dtype=np.int_, delimiter=",",
+    ...                         skip_header=7, usecols=range(3, 17))
+    >>> dates = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
+    ...                       max_rows=1, usecols=range(3, 17))
+    >>> locations = np.genfromtxt(filename, dtype=np.unicode_, delimiter=",",
+    ...                           skip_header=7, usecols=(0, 1))
+
+Included in the :func:`numpy.genfromtxt` function call, we have selected the
+:class:`numpy.dtype` for each subset of the data (either an integer -
+:class:`numpy.int_` - or a string of characters - :class:`numpy.unicode_`). We
+have also used :doc:`slice <basics.indexing>` syntax to select the right
+columns for the data we are interested in. You can read more about the
+:func:`numpy.genfromtxt` function from the :func:`Reference Documentation
+<numpy.genfromtxt>` or from the :doc:`Basic IO tutorial <basics.io.genfromtxt>`.
+
+**Exploring the data**
+
+First of all, we can plot the whole set of data we have and see what it looks
+like. In order to get a readable plot, we select only a few of the dates to
+show in our :func:`x-axis ticks <matplotlib.pyplot.xticks>`. Note also that in
+our plot command, we use ``nbcases.T`` (the transpose of the ``nbcases`` array)
+since this means we will plot each row of the file as a separate line. We choose
+to plot a dashed line (using the ``'--'`` line style). See the
+`matplotlib <https://matplotlib.org/>`_ documentation for more info on this.
+
+.. code::
+
+    >>> import matplotlib.pyplot as plt
+    >>> plt.plot(dates, nbcases.T, '--')
+    >>> selected_dates = [0, 3, 11, 13]
+    >>> plt.xticks(selected_dates, dates[selected_dates])
+    >>> plt.title("COVID-19 cumulative cases from Jan 21 to Feb 3 2020")
+    
+.. plot:: user/plot_covid_1.py
+    :align: center
+    :include-source: 0
```

Comment:
I'm a little worried that these python files and this rst file are going to get out of sync. @mattip, do you know of a way to get all of the code in the rst, but split it across multiple plots?

Reply:
I changed this document to use the IPython sphinx directive.